### PR TITLE
[Data Collection] Add support for persisting geometry data in local and remote db

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards https://github.com/google/ground-android/issues/1351

The main goal of this PR is to fix data collection flow for point and polygon type tasks. This requires persisting the geometry as a mutation before it can be synced to firestore. As a bonus, this also adds support for creating polygons in the suggest LOI flow.

List of all things done in this PR:
- Adds a common wrapper class `GeometryData` for geometry tasks
- Replaces usage of `DropAPinTaskData` with `GeometryData`
- Save geometry created in DrawPolygon task to response map
- Adds support for creating polygon type LOIs in `SuggestLoi` flow
- Persist geometry in response to local db and remote firestore db
- Add unit tests

Verified the changes by creating a submission which contains "drop a pin" and "draw polygon" tasks and checking the results in firebase. Also, the submission counter in the mobile gets incremented afterwards which means that the entire roundtrip flow is working as expected.

@JSunde @gino-m  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/model/submission/GeometryData.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.google.android.ground.model.geometry.LinearRing
+import com.google.android.ground.model.geometry.MultiPolygon
+import com.google.android.ground.model.geometry.Point
+import com.google.android.ground.model.geometry.Polygon
+import java8.util.Optional
+
+/** A user provided response to a geometry based task. */
+data class GeometryData(val geometry: Geometry) : TaskData {
+
+  // TODO: Fix this!!!
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/model/submission/GeometryData.kt:29]: +1. Created https://github.com/google/ground-android/issues/1733 to track. Let's mention here?

File: ground/src/main/java/com/google/android/ground/model/submission/GeometryData.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      is Polygon -> "Polygon data"
+      is LinearRing -> "LinearRing data"
+      is LineString -> "LineString data"
+      is MultiPolygon -> "MultiPolygon data"
+    }
+
+  override fun isEmpty(): Boolean = false
+
+  companion object {
+    fun fromGeometry(geometry: Geometry?): Optional<GeometryData> =
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/model/submission/GeometryData.kt:42]: Can we replace the return values with `GeometryData?` while we're at it?

File: ground/src/test/java/com/google/android/ground/persistence/remote/firebase/schema/SubmissionMutationConverterTest.kt [hunk 3]
```diff
@@ -15,25 +15,34 @@
...
 import com.sharedtest.FakeData
 import java.util.Date
+import java8.util.Optional
 import kotlinx.collections.immutable.persistentListOf
 import org.junit.Assert.assertThrows
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.mockito.junit.MockitoJUnitRunner
 
+// TODO(Shobhit): Add coverage for date, time task types
```
Comment (gino-m) [inline ground/src/test/java/com/google/android/ground/persistence/remote/firebase/schema/SubmissionMutationConverterTest.kt:45]: Added https://github.com/google/ground-android/issues/1734, please reference here.

**Review by gino-m — CHANGES_REQUESTED** [review]
Glad to see this fixed, great work @shobhitagarwal1612 . A few nits, otherwise LGTM.

File: ground/src/main/java/com/google/android/ground/model/submission/GeometryData.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/model/submission/GeometryData.kt]: Thanks for filing the bug. Linked.

File: ground/src/main/java/com/google/android/ground/model/submission/GeometryData.kt (hunk 2 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/model/submission/GeometryData.kt:42]: Filed a bug #1735 and added a todo to perform this cleanup separately.

File: ground/src/test/java/com/google/android/ground/persistence/remote/firebase/schema/SubmissionMutationConverterTest.kt (hunk 3 above)
Comment (shobhitagarwal1612) [reply ground/src/test/java/com/google/android/ground/persistence/remote/firebase/schema/SubmissionMutationConverterTest.kt]: Done

**Review by gino-m — APPROVED** [review]
🥳

**Pull request merged**